src/IRrecv.cpp

```diff
@@ -601,25 +606,29 @@ bool IRrecv::decodeHash(decode_results *results) {
 //   onespace:  Nr. of uSeconds in an expected space signal for a '1' bit.
 //   zeromark:  Nr. of uSeconds in an expected mark signal for a '0' bit.
 //   zerospace: Nr. of uSeconds in an expected space signal for a '0' bit.
+//   tolerance: Percentage error margin to allow.
 // Returns:
 //  A match_result_t structure containing the success (or not), the data value,
 //  and how many buffer entries were used.
-match_result_t IRrecv::matchData(volatile uint16_t *data_ptr, uint16_t nbits,
-                                 uint16_t onemark, uint32_t onespace,
-                                 uint16_t zeromark, uint32_t zerospace) {
+match_result_t IRrecv::matchData(volatile uint16_t *data_ptr,
+                                 const uint16_t nbits, const uint16_t onemark,
+                                 const uint32_t onespace,
+                                 const uint16_t zeromark,
+                                 const uint32_t zerospace,
+                                 const uint8_t tolerance) {
```

Comment:
With the addition of the new `tolerance` option, should this default to the constant?

Reply:
It has a default set in the corresponding header file. Otherwise, all the other calls to it would a) not compile, and b) fail their unit-tests. ;-)

See https://github.com/markszabo/IRremoteESP8266/pull/359/files#diff-1e400ae20a0050642418d25a07ef66a0R142

Reply:
I should say also. Thanks for picking it up, even if it is a false-positive issue. I'd rather double check something then have to make a subsequent patch/break the build etc.